pyccel/parser/semantic.py

```diff
@@ -2304,12 +2309,23 @@ def _visit_FunctionDef(self, expr, **settings):
         templates.update(expr.templates)
 
         if cls_name:
-            header += self.get_header(cls_name +'.'+ name)
+            tmp_headers = self.get_header(cls_name + '.' + name)
+            for header in tmp_headers:
+                if all(header.dtypes != hd.dtypes for hd in headers):
+                    headers.append(header)
+                else:
+                    errors.report(DUPLICATED_SIGNATURE, symbol=header,
+                            severity='warning')
```

Comment:
Is there a reason why this bit needs to be duplicated in the if/else block? I can't see a difference between the two